src/redis-check-rdb.c

```diff
@@ -205,13 +205,15 @@ int redis_check_rdb(char *rdbfilename, FILE *fp) {
     static rio rdb; /* Pointed by global struct riostate. */
     struct stat sb;
 
-    if (isFifo(rdbfilename)) {
-        /* Cannot check RDB over named pipe because fopen blocks until another process opens the FIFO for writing. */
-        return 1;
-    }
-
     int closefile = (fp == NULL);
-    if (fp == NULL && (fp = fopen(rdbfilename,"r")) == NULL) return 1;
+    if (fp == NULL) {
+        if (isFifo(rdbfilename)) {
```

Comment:
how about changing things so that we can also handle a case were someone used standalone redis-check-rdb executable on a fifo? i.e. it's a minor change of using `fstat` instead of `stat` (and remembering to close the file on exit)?

Reply:
or actually now that i think of it, maybe the problem is not at all in redis-check-rdb, it can probably operate on a FIFO fine. the problem might actually be in rdb.c which should have avoided calling `redis_check_rdb()` in that case (i.e. the problem is the attempt to reuse the FIFO).
am i right?

Reply:
actually, i think i mixed some things up (AOF check and replication), but actually let's take a step backwards.
what exactly is the problem with operating on a FIFO? we don't do `seek()` anyway, right?
i imagine the only problem (the hang you mentioned in the previous PR) is the attempt to re-open a "file" for the second time.
which again means the problem is at the caller (not in redis-check-rdb.c itself)